src/foam/dashboard/model/Metric.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dashboard.model',
+  name: 'Metric',
+  extends: 'foam.dashboard.model.Visualization',
+
+  requires: [
+    'foam.dashboard.view.Metric as MetricView',
+    'foam.mlang.sink.Count',
+    'foam.mlang.sink.Sum',
+    'foam.mlang.sink.Min',
+    'foam.mlang.sink.Max',
+    'foam.mlang.sink.Average',
+    'foam.mlang.predicate.False',
+    'foam.parse.QueryParser',
+    'foam.u2.ContextSensitiveDetailView as DetailView'
+  ],
+
+  properties: [
+    {
+      name: 'views',
+      factory: function() {
+        return [
+          [ this.MetricView,  'Metric' ],
+          [ this.DetailView, 'Configure' ]
+        ];
+      }
+    },
+    {
+      class: 'Enum',
+      of: 'foam.dashboard.model.MetricOperation',
+      name: 'operation',
+      value: 'COUNT'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'property',
+      documentation: 'Property to calculate metric on (for SUM, MIN, MAX, AVG operations)'
+    },
+    {
+      name: 'sink',
+      factory: function() {
+        return this.createSink();
+      }
+    }
+  ],
+
+  methods: [
+    function createSink() {
+      var ExpressionsSingleton = foam.mlang.ExpressionsSingleton.create();
+      
+      switch(this.operation) {
```

Comment:
You can have methods on Enum values (see docs in Enum.js), so it would be nicer if here you just did this.operation.createSink(); Then when you added new operations you would only need to update in one place.

Reply:
done.